src/core/math/vec2.js

```diff
@@ -407,6 +409,57 @@ class Vec2 {
         return this;
     }
 
+    /**
+     * Rotate a vector by Euler angle.
+     *
+     * @param {number} degrees - The number to rotate by Euler angle.
+     * @returns {Vec2} Self for chaining.
+     * @example
+     * const v = new pc.Vec2(0, 10);
+     *
+     * v.rotate(45); // rotates by 45 degrees
+     *
+     * // Outputs [7.071068.., 7.071068..]
+     * console.log("Vector after rotation is: " + v.toString());
+     */
+    rotate(degrees) {
+        const angle = Math.atan2(this.x, this.y) + (degrees / math.RAD_TO_DEG);
```

Comment:
how about
```
const angle = (this.angle() + degrees) / math.RAD_TO_DEG;
```

Even though I understand you're trying to avoid a function call.

Reply:
or even better
```
const angle = (this.angle() + degrees) * math.DEG_TO_RAD;
```

Reply:
actually, I think I like your code better, avoid the function and multiply. Just replace the / by *math.DEG_TO_RAD I think.